auction-server/src/opportunity/repository/mod.rs

```diff
@@ -24,12 +26,17 @@ mod refresh_in_memory_opportunity;
 mod remove_opportunities;
 mod remove_opportunity;
 
-pub use models::*;
+pub use {
+    db::*,
+    models::*,
+};
+
 pub const OPPORTUNITY_PAGE_SIZE_CAP: usize = 100;
 
 #[derive(Debug)]
-pub struct Repository<T: InMemoryStore> {
+pub struct Repository<T: InMemoryStore, U: OpportunityTable<T> = DB> {
     pub in_memory_store: T,
+    pub db:              U,
```

Comment:
I think it's probably worth it using dynamic dispatch here to avoid the generic `U` that appears in a lot of places now